pentasharp/Interfaces/ITaxiReservationService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using pentasharp.Models.TaxiRequest;
+using pentasharp.ViewModel.TaxiReservation;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace pentasharp.Interfaces
+{
+    /// <summary>
+    /// Defines the contract for managing taxi reservations.
+    /// </summary>
+    public interface ITaxiReservationService
+    {
+        /// <summary>
+        /// Searches for available taxi companies.
+        /// </summary>
+        /// <returns>A task representing an asynchronous operation that returns a collection of available taxi companies as DTOs.</returns>
```

Comment:
this can be changed to 
/// <returns>The <see cref="TaxiCompanyRequest"/> model. </returns>, do the similar approach to all other return in the summaries here